src/common/date-helpers.ts

```diff
@@ -0,0 +1,48 @@
+export function getDayNames (firstDay: number = 0, nameLength: string = 'short', locale: string = 'en-US'): Array<string> {
+    // Days start from Sunday (Sunday = 0, Monday = 1, etc.)
+    const listOfDayNames: Array<string> = [];
+
+    // For each of the seven days of the week
+    for (let i = 0; i < 7; i++) {
+        let dayName = new Date(1970, 5, i + firstDay).toLocaleString(locale, { weekday: nameLength });
+        listOfDayNames.push(dayName);
+    }
+
+    return listOfDayNames;
+}
```

Comment:
it is subjective, yes: or you using imperative style, or functional. 
`.map` and chaining are abstractions made to avoid loops. It is fact. Use this abstraction or not is subjective. 
I think if language gives you abstraction you have to it.